### PR TITLE
Adding support for Openshift Security Context Constraints

### PR DESCRIPTION
**Pull request by sjmiller609**
## Description

Peer programmed with @ianstanton 

Adding support for Openshift Security Context Constraints

## 🎟 Issue(s)

Part of https://github.com/astronomer/issues/issues/2039

## 🧪  Testing

This configuration is default disabled. For the enabled case, a unit test has been added.

## 📋 Checklist

- [x] The PR title is informative to the user experience
- [ ] Functional test(s) added
- [x] Helm chart unit test(s)

### CODE REVIEW

File: templates/airflow-scc-anyuid.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+fsGroup:
+  type: RunAsAny
+groups:
+- system:cluster-admins
+allowHostDirVolumePlugin: false
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPID: false
+allowHostPorts: false
+allowPrivilegeEscalation: true
```
Comment (schnie) [inline templates/airflow-scc-anyuid.yaml:41]: Does this need to be `true` for something to work?

Comment (ianstanton) [reply templates/airflow-scc-anyuid.yaml:41]: @sjmiller609 and I confirmed we do not need `allowPrivilegeEscalation: true`. We'll remove this.

File: templates/airflow-scc-anyuid.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- downwardAPI
+- emptyDir
+- persistentVolumeClaim
+- projected
+- secret
+defaultAddCapabilities: null
+fsGroup:
+  type: RunAsAny
+groups:
+- system:cluster-admins
```
Comment (schnie) [inline templates/airflow-scc-anyuid.yaml:35]: What does this `groups` section do?

Comment (ianstanton) [reply templates/airflow-scc-anyuid.yaml:35]: We likely don't need this `groups` section as we have a list of users above. Some info on groups (a collection of users) https://docs.openshift.com/container-platform/3.9/admin_guide/manage_users.html#admin-guide-managing-users-creating-groups

File: templates/airflow-scc-anyuid.yaml (hunk 1 above)
Comment (ianstanton) [reply templates/airflow-scc-anyuid.yaml:39]: We found this needs to be true (at least with Debian image). Without this, we receive the following error:

```
▶ oc logs airflow-scheduler-7d46df6bb6-28lst -n airflow -c wait-for-airflow-migrations
error: failed switching to "astro": operation not permitted
```

File: templates/airflow-scc-anyuid.yaml (hunk 2 above)
Comment (ianstanton) [reply templates/airflow-scc-anyuid.yaml]: Found that this is not necessary. Removed this section.

**Review by ianstanton — APPROVED** [review]

File: Chart.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # apiVersion v1 is Helm 2
 apiVersion: v1
 name: airflow
-version: 0.18.0
+version: 0.18.1
```
Comment (ashb) [inline Chart.yaml:4]: I don't think 0.18.0 has actually been released yet (but I'm not 100% sure) -- we're still testing changes for Airflow 2.0 here.

Comment (sjmiller609) [inline Chart.yaml:4]:
```suggestion
version: 0.18.0
```

**Pull request merged**